tests/HUBActionRegistryTests.m

```diff
@@ -57,8 +57,9 @@ - (void)testRegisteringFactoryAndCreatingAction
     HUBActionMock * const action = [[HUBActionMock alloc] initWithBlock:nil];
     HUBActionFactoryMock * const factory = [[HUBActionFactoryMock alloc] initWithActions:@{actionIdentifier.namePart: action}];
     [self.actionRegistry registerActionFactory:factory forNamespace:actionIdentifier.namespacePart];
-    
-    XCTAssertEqual([self.actionRegistry createCustomActionForIdentifier:actionIdentifier], action);
+
+    // These should be the same instance.
+    XCTAssertTrue([self.actionRegistry createCustomActionForIdentifier:actionIdentifier] == action);
```

Comment:
How about adding a description to the assert? That is:
```objc
XCTAssertTrue([self.actionRegistry createCustomActionForIdentifier:actionIdentifier] == action, @"The action should be the same instance as the factory mock vends");
```

That way the comment will be directly attached to the assert. Making sure it also shows up when the test fails.